sdk/provisioning/Azure.Provisioning/src/storage/StorageAccount.cs

```diff
@@ -50,5 +45,32 @@ public StorageAccount(IConstruct scope, StorageKind kind, StorageSkuName sku, Re
             }
             return result;
         }
+
+        /// <inheritdoc/>
+        protected override string GetAzureName(IConstruct scope, string resourceName)
+        {
+            var span = resourceName.AsSpan();
+            StringBuilder stringBuilder = new StringBuilder();
+            for (int i = 0; i < span.Length; i++)
+            {
+                char c = span[i];
+                if (!char.IsLetterOrDigit(c))
+                {
+                    continue;
+                }
+                if (char.IsDigit(c) && !char.IsLower(c))
+                {
+                    stringBuilder.Append(char.ToLower(c));
+                }
+                else
+                {
+                    stringBuilder.Append(c);
+                }
```

Comment:
I think there's a logic error here and you'll never actually lowercase any letters.